src/console/discovery.ts

```diff
@@ -0,0 +1,151 @@
+import dgram from "dgram";
+import _ from "lodash";
+import moment from "moment";
+
+import { discoverConsoleFound, discoverConsoleLost } from "./ipc";
+
+export interface DiscoveredConsoleInfo {
+  ip: string;
+  mac: string;
+  name: string | undefined;
+  timeout: NodeJS.Timeout;
+  firstFound: moment.Moment;
```

Comment:
Again, since `DiscoveredConsoleInfo` is sent to the renderer via IPC, the JSON we send should be serializable. `timeout` shouldn't be sent to the frontend, and `firstFound` should probably be an ISO string.